tests/simulator/py/refined_particle_nbr.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+#
+# formatted with black
+
+from phare import cpp
+
+import unittest, os, phare.pharein as ph
+from datetime import datetime, timezone
+from tests.diagnostic import dump_all_diags
+from tests.simulator.py import create_simulator
+from phare.data.wrangler import DataWrangler
+
+out = "phare_outputs/refined_particle_nbr"
+diags = {"diag_options": {"format": "phareh5", "options": {"dir": out}}}
+
+
+class SimulatorRefinedParticleNbr(unittest.TestCase):
+    """
+      refine 10 cells in 1d, ppc 100, interp 3
+        10 * 2 * ppc = 200
+        10 * 3 * ppc = 300 300 / 200 = 1.5
+        10 * 4 * ppc = 400 500 / 400 = 1.33
+        10 * 5 * ppc = 500 500 / 400 = 1.25
+      taking the minimul diff across permutations
+      current to previous should be at least this times bigger
+    """
+    PREVIOUS_ITERATION_DIFF_1d = 1.25
+
+    def dup(dic):
+        dic.update(diags.copy())
+        dic.update({"diags_fn": lambda model: dump_all_diags(model.populations)})
+        return dic
+
+    def tearDown(self):
+        for k in ["dman", "sim", "hier"]:
+            if hasattr(self, k):
+                v = getattr(self, k)
+                del v  # blocks segfault on test failure, could be None
+        cpp.reset()
+
+    def _do_dim(self, dim, input, diff):
+        from phare.pharein.simulation import valid_refined_particle_nbr
+
+        for interp in range(1, 4):
+            previous_split_particle_max = 0
+            for refined_particle_nbr in valid_refined_particle_nbr[dim][interp]:
+                input["refined_particle_nbr"] = refined_particle_nbr
+                self.dman, self.sim, self.hier = create_simulator(dim, interp, **input)
+                self.dman.dump(self.sim.currentTime(), self.sim.timeStep())
+                dw = DataWrangler(self.sim, self.hier)
+                max_per_pop = 0
+                for pop, particles in dw.getPatchLevel(1).getParticles().items():
+                    per_pop = 0
+                    nCells = 0
+                    for key, patches in particles.items():
+                        for patch in patches:
+                            nCells += int(patch.cells.split(",")[0])
+                            per_pop += patch.data.size()
+                max_per_pop = max(max_per_pop, per_pop)
+                # splitting is not perfect, so we remove nCells to counter
```

Comment:
I don't understand the comment, so neither the -nCells

Reply:
I think it's to account for this: 

https://github.com/PHAREHUB/PHARE/blob/master/src/amr/data/particles/refine/particles_data_split.h#L174

"max_per_pop" comes out less than the expected "PREVIOUS_ITERATION_DIFF_1d"

assuming I've not made a mistake somewhere

Reply:
I don't understand. The loop you're pointing at is the loop over the different directions that calculates the mother particle position in the refined grid frame.
If yo have a domain, say from cell 20 to cell 25 that you have to fill from coarse particles. You have to split, say N coarse particles, essentially positionned "under" cells 20-25. (say 10-12 for ratio 2). Each of those N particle will split M daughter particles, so you expect N*M refined particles. If most of these N*M fall in refined cells 20-25, some (those coming from mother particles close to the borders) will fall outside. As a result I would not be surprised to have less than N*M refined particles... it that what you think you got ?

Reply:
That sure sounds like it.

Reply:
you could check that, discarding the splitting of the mother on border cells (those are the only one suceptible to split daughters outside) you get exactly N*M

Reply:
sounds good